overlord/snapstate/snapstate.go

```diff
@@ -462,20 +462,66 @@ func UpdateMany(st *state.State, names []string, userID int) ([]string, []*state
 		}
 	}
 
-	updated := make([]string, 0, len(updates))
+	params := func(update *snap.Info) (string, Flags, *SnapState) {
```

Comment:
because one takes channel from snapst and the other case takes it on the side for params so I need some function update or snapst -> channel, flags anyway